jack.py

```diff
@@ -1533,6 +1562,65 @@ def get_array(self):
         return np.frombuffer(self.get_buffer(), dtype=np.float32)
 
 
+def _maybe_bytes(data):
+    if type(data) is bytes:
+        data_ptr = _ffi.new("unsigned char[]", len(data))
+        data_ptr[0:len(data)] = data
+        return data_ptr
+    return data
+
+def _returns_bytes(fun):
+    def wrapper(self, cnt):
+        ret_cnt, data = fun(self, cnt)
+        return bytes(_ffi.buffer(data, ret_cnt))
+    return wrapper
+
+class RingBuffer(object):
+    def __init__(self, size=256):
+        ptr = _lib.jack_ringbuffer_create(size + 1)
+        if not ptr:
+            raise JackError(
+                "Could not create ringbuffer of size {0}".format(size))
+        self._ptr = _ffi.gc(ptr, _lib.jack_ringbuffer_free)
+
+    def _read(self, cnt):
+        data = _ffi.new("unsigned char[]", cnt)
+        ret_cnt = _lib.jack_ringbuffer_read(self._ptr, data, cnt)
+        return ret_cnt, data
+    read = _returns_bytes(_read)
+
+    def _peek(self, cnt):
+        data = _ffi.new("unsigned char[]", cnt)
+        ret_cnt = _lib.jack_ringbuffer_peek(self._ptr, data, cnt)
+        return ret_cnt, data
+    peek = _returns_bytes(_peek)
+
+    def write(self, data):
+        data = _maybe_bytes(data)
+        return _lib.jack_ringbuffer_write(self._ptr, data, _ffi.sizeof(data))
+
+    def mlock(self):
+        return _lib.jack_ringbuffer_mlock(self._ptr)
+
+    def reset(self):
+        _lib.jack_ringbuffer_reset(self._ptr)
+
+    @property
+    def read_space(self):
+        return _lib.jack_ringbuffer_read_space(self._ptr)
+
+    @property
+    def write_space(self):
+        return _lib.jack_ringbuffer_write_space(self._ptr)
+
+    @property
+    def size(self):
+        return self._ptr.size - 1
+
+    def __repr__(self):
+        return "<jack.RingBuffer {0.read_space}/{0.size}>".format(self)
```

Comment:
I think having `space occupied` / `total space` is a little more in line to how people expect it to be. You can even do the division to get the percentage occupied.

If we change it then '/' might not make sense anymore. I guess we could use something like "<->". There will also be confusion on what's occupied and what's available.

The awkwardness is going to be there either way, write space at the beginning might still be bigger than the size argument.